server_python_scripts/blocks_base_classes/opmode.py

```diff
@@ -7,6 +7,9 @@ def __init__(self, robot: RobotBase):
     def start(self) -> None:
         self.robot.start()
     def loop(self) -> None:
+        # Call steps method if it exists in the derived class
+        if hasattr(self, 'steps'):
```

Comment:
We should make sure it's callable.
`if hasattr(self, 'steps') and callable(self.steps):`